api/src/main/java/jakarta/data/metamodel/restrict/TextRestrictionRecord.java

```diff
@@ -17,76 +17,46 @@
-        String value) implements TextRestriction<T> {
+record TextRestrictionRecord<T>(String attribute, Like constraint, boolean negated)
+        implements TextRestriction<T> {
 
     TextRestrictionRecord {
         Objects.requireNonNull(attribute, "Attribute must not be null");
+        Objects.requireNonNull(constraint, "Pattern must not be null");
     }
 
-    TextRestrictionRecord(String attributeName, Operator comparison, boolean escaped, String value) {
-        this(attributeName, comparison, true, escaped, value);
+    public TextRestrictionRecord(String attribute, Like range) {
+        this(attribute, range, false);
     }
 
-    TextRestrictionRecord(String attributeName, Operator comparison, String value) {
-        this(attributeName, comparison, true, false, value);
+    @Override
+    public Operator comparison() {
+        return negated ? constraint.operator().negate() : constraint.operator();
     }
 
     @Override
-    public TextRestriction<T> ignoreCase() {
-        return new TextRestrictionRecord<>(attribute, comparison, false, isEscaped, value);
+    public TextRestrictionRecord<T> negate() {
+        return new TextRestrictionRecord<>(attribute, constraint, !negated);
     }
 
     @Override
-    public TextRestriction<T> negate() {
-
-        return new TextRestrictionRecord<>(
-                attribute,
-                comparison.negate(),
-                isCaseSensitive,
-                isEscaped,
-                value);
+    public TextRestrictionRecord<T> ignoreCase() {
+        return new TextRestrictionRecord<>(attribute, constraint.ignoreCase(), negated);
     }
 
     /**
-     * Textual representation of a text restriction.
+     * Textual representation of a basic restriction.
      * For example,
-     * <pre>name LIKE "Jakarta EE %" IGNORE_CASE</pre>
+     * <pre>price < 50.0</pre>
      *
-     * @return textual representation of a text restriction.
+     * @return textual representation of a basic restriction.
      */
     @Override
     public String toString() {
-        String comparisonString = comparison.asQueryLanguage();
-        String valueString = value == null ? "null" : value;
-        StringBuilder builder = new StringBuilder(
-                attribute.length() +
-                comparisonString.length() +
-                valueString.length() +
-                24); // number of additional characters that might be appended
-        builder.append(attribute).append(' ')
-               .append(comparisonString).append(' ');
-        if (value == null) {
-            builder.append(valueString);
-        } else {
-            builder.append('"').append(valueString).append('"');
-        }
-        if (isEscaped) {
-            builder.append(" ESCAPED");
-        }
-        if (!isCaseSensitive) {
-            builder.append(" IGNORE_CASE");
-        }
-        return builder.toString();
+        return attribute + ' ' + comparison().asQueryLanguage() + ' ' + constraint;
```

Comment:
Now I see why some constraint records were only printing the value and not the operation.  I proposed change it so that a user will see a more accurate and meaningful output when invoking toString on a Constraint.
```suggestion
        return attribute + ' ' + constraint;
```